server/src/test/java/io/crate/planner/optimizer/symbol/rule/SimplifyEqualsOperationOnIdenticalReferencesTest.java

```diff
@@ -188,12 +170,62 @@ public void testSimplifyRefEqRefWhenParentIsIgnore3vlFunction() {
                     DataTypes.BOOLEAN)
             ),
             DataTypes.BOOLEAN);
-        Match<Function> match = RULE.pattern().accept(func.arguments().get(0), Captures.empty());
-        Symbol optimizedFunction = null;
-        if (match.isPresent()) {
-            optimizedFunction = RULE.apply(match.value(), match.captures(), NODE_CONTEXT, func);
-        }
-        Symbol expected = Literal.BOOLEAN_TRUE;
-        assertThat(optimizedFunction, is(expected));
+        Symbol optimizedFunction = OPTIMIZER.optimize(functionToOptimize);
+        assertThat(optimizedFunction.toString(), is("ignore3vl(true)"));
+    }
+
+    @Test
+    public void testSimplifyRefEqRefWhenParentIsNotIgnore3vlFunction() {
+        var refEqRef = new Function(
+            EqOperator.SIGNATURE,
+            List.of(
+                NOT_NULLABLE_REF,
+                NOT_NULLABLE_REF
+            ),
+            DataTypes.BOOLEAN);
+        var not = new Function(
+            NotPredicate.SIGNATURE,
+            List.of(refEqRef),
+            DataTypes.BOOLEAN
+        );
+        Function functionToOptimize = new Function(
+            Ignore3vlFunction.SIGNATURE,
+            List.of(not),
+            DataTypes.BOOLEAN);
+        Symbol optimizedFunction = OPTIMIZER.optimize(functionToOptimize);
+        assertThat(optimizedFunction.toString(), is("ignore3vl((NOT true))"));
+    }
+
+    @Test
+    public void testMultipleApplicationOfSimplifyRefEqRef() {
```

Comment:
this implicitly tests `Optimizer.Visitor`. If the previously visited `Function` is not properly popped, `right` will be simplified to `true`.